src/utils.py

```
@@ -8,122 +9,275 @@
+        file_var = f"{env_var}_FILE"
+        file_path = os.getenv(file_var)
+        if file_path and os.path.exists(file_path):
+            try:
+                with open(file_path, 'r') as f:
+                    return f.read().strip()
+            except Exception as e:
+                logger.error(f"Failed to read credential file {file_path}: {e}")
+        
+        return None
+
+@dataclass 
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)